### PR TITLE
R2 support

### PR DESCRIPTION
**Pull request by ohmpatel1997**
**Summary of changes**
Changes introduced in this pull request:
Adds a functionality to existing CF worker to fetch the archive snapshot via path:

	{ pattern = "forest-archive.chainsafe.dev/archive/calibnet/*", zone_name = "chainsafe.dev" },
	{ pattern = "forest-archive.chainsafe.dev/archive/mainnet/*", zone_name = "chainsafe.dev" },
	{ pattern = "forest-archive.chainsafe.dev/archive/historical/*", zone_name = "chainsafe.dev" }
-

**Reference issue to close (if applicable)**

Closes

**Other information and links**

### CODE REVIEW

**Review by aatifsyed — APPROVED** [review]

File: cf/latest-snapshot/src/index.ts [hunk 1]
```diff
@@ -8,92 +9,139 @@ function basename(path: string) {
...
 		// Disallow any other request method except HEAD and GET, they are not sensible in the context
 		// of fetching a snapshot.
 		switch (request.method) {
-			case 'HEAD':
-			case 'GET':
-				return await get_latest(request.headers, env, chain);
-			default:
+			case 'HEAD': {
+				break;
+			}
```
Comment (lemmih) [inline cf/latest-snapshot/src/index.ts:120]: Is this the correct behavior for HEAD requests?

File: cf/latest-snapshot/README.md [hunk 2]
```diff
@@ -5,6 +5,10 @@ This worker acts on two endpoints:
 - `https://forest-archive.chainsafe.dev/latest/calibnet/`
 - `https://forest-archive.chainsafe.dev/latest/mainnet/`
 
+- `https://forest-archive.chainsafe.dev/archive/calibnet/*`
+- `https://forest-archive.chainsafe.dev/archive/mainnet/*`
+- `https://forest-archive.chainsafe.dev/archive/historical/*`
```
Comment (LesnyRumcajs) [inline cf/latest-snapshot/README.md:10]: It's no longer `two endpoints` mentioned in line 3, no?

File: cf/latest-snapshot/README.md
Comment (LesnyRumcajs) [inline cf/latest-snapshot/README.md:1]: Is the name `latest-snapshot` still suitable for this worker?

File: cf/latest-snapshot/src/index.ts (hunk 1 above)
Comment (LesnyRumcajs) [reply cf/latest-snapshot/src/index.ts:120]: Previosuly it was a fallthrough to get the same code path as GET, now we're breaking without any actions. I don't think it's correct, unless I'm missing something.

**Review by lemmih — APPROVED** [review]

**Pull request merged**